toolchain/sem_ir/function.h

```diff
@@ -15,6 +15,9 @@ namespace Carbon::SemIR {
 
 // Function-specific fields.
 struct FunctionFields {
+  // Kinds of special functions.
+  enum class SpecialFunctionKind : uint8_t { None, Thunk };
```

Comment:
What other special functions are you anticipating, that this isn't a bool?

Reply:
There are some other kinds used [in clang](https://github.com/llvm/llvm-project/blob/231aa3070dcd91e10e9972d20f7557c0068c41e3/clang/include/clang/AST/GlobalDecl.h): things like multi-version dispatchers and host -> device trampolines come up. C++ also has multiple symbols for destructor variants, but I think we'll end up modeling that in a different way. For now this is a little speculative, though.